flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java

```diff
@@ -105,12 +105,14 @@ public boolean equals(Object o) {
 
         SavepointRestoreSettings that = (SavepointRestoreSettings) o;
         return allowNonRestoredState == that.allowNonRestoredState
-                && (Objects.equals(restorePath, that.restorePath));
+                && Objects.equals(restorePath, that.restorePath)
+                && Objects.equals(restoreMode, that.restoreMode);
     }
 
     @Override
     public int hashCode() {
         int result = restorePath != null ? restorePath.hashCode() : 0;
+        result = 31 * result + (restoreMode != null ? restoreMode.hashCode() : 0);
```

Comment:
The restoreMode property in SavepointRestoreSettings should be @Nonnull ?

Reply:
Thanks for the suggestion, I think it's reasonable to mark it as '@ Nonnull'